tech/tech/api/products_router.py

```
@@ -0,0 +1,132 @@
+    Raises:
+        HTTPException: If a product with the same name already exists.
+    """
+    use_case = CreateProductUseCase(repo)
+    try:
+        created_product = use_case.execute(product)
+        return created_product
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
+
+@router.get('/{category}', response_model=list[ProductPublic])
+def list_products_by_category(category: str, repo: SQLAlchemyProductRepository = Depends(get_product_repository)):
```

Ruff (0.8.2)

Line too long (114 > 79)

(E501)